### PR TITLE
fix(search): valueChange

### LINKED ISSUE #651: Search Component: clear button doesn't have separate event
Right now, Search have same change event emitter for typing and cross button.
So, it is not possible to detect where user click the cross button or simply empty the text field manually because in both the scenarios, search is emitting empty field

Expectation: 
Separate event emitter for cross button click or something to identify separately.

### PR DESCRIPTION
**Pull request by youda97**
closes #651

### CODE REVIEW

File: src/search/search.component.ts [hunk 1]
```diff
@@ -11,20 +11,6 @@ import {
 import { NG_VALUE_ACCESSOR, ControlValueAccessor } from "@angular/forms";
 import { I18n } from "../i18n/i18n.module";
 
-/**
- * Used to emit changes performed on search components.
- */
-export class SearchChange {
```
Comment (cal-smith) [inline src/search/search.component.ts]: We can't remove this just yet ... we can definitely mark it deprecated however :+1:

File: src/search/search.component.ts [hunk 2]
```diff
@@ -125,9 +111,13 @@ export class Search implements ControlValueAccessor {
 	 */
 	@Input() clearButtonTitle = this.i18n.get().SEARCH.CLEAR_BUTTON;
 	/**
-	 * Emits event notifying other classes when a change in state occurs in the input.
+	 * Emits an event when value is changed.
 	 */
-	@Output() change = new EventEmitter<SearchChange>();
```
Comment (cal-smith) [inline src/search/search.component.ts]: Same as above, we'll need to keep this as deprecated until a major version

File: src/search/search.component.ts [hunk 3]
```diff
@@ -125,9 +126,18 @@ export class Search implements ControlValueAccessor {
...
 	@Input() clearButtonTitle = this.i18n.get().SEARCH.CLEAR_BUTTON;
 	/**
+	 * @deprecated in the next major carbon-components-angular version in favour of `valueChange`
 	 * Emits event notifying other classes when a change in state occurs in the input.
 	 */
 	@Output() change = new EventEmitter<SearchChange>();
+	/**
+	 * Emits an event when value is changed.
+	 */
+	@Output() valueChange = new EventEmitter<any>();
```
Comment (cal-smith) [inline src/search/search.component.ts]: In this case we know what types we're dealing with ...

```suggestion
	@Output() valueChange = new EventEmitter<string>();
```

File: src/search/search.component.ts [hunk 4]
```diff
@@ -12,6 +12,7 @@ import { NG_VALUE_ACCESSOR, ControlValueAccessor } from "@angular/forms";
 import { I18n } from "../i18n/i18n.module";
 
 /**
+ * @deprecated in the next major carbon-components-angular version in favour of `valueChange`
```
Comment (zvonimirfras) [inline src/search/search.component.ts]:
```suggestion
 * @deprecated in favor of `valueChange`, to be removed in the next major carbon-components-angular version
```

File: src/search/search.component.ts [hunk 5]
```diff
@@ -125,9 +126,18 @@ export class Search implements ControlValueAccessor {
 	 */
 	@Input() clearButtonTitle = this.i18n.get().SEARCH.CLEAR_BUTTON;
 	/**
+	 * @deprecated in the next major carbon-components-angular version in favour of `valueChange`
```
Comment (zvonimirfras) [inline src/search/search.component.ts]:
```suggestion
	 * @deprecated in favor of `valueChange`, to be removed in the next major carbon-components-angular version
```

**Pull request merged**